internal/discovery/pods_lister.go

```diff
@@ -0,0 +1,34 @@
+package discovery
+
+import (
+	"time"
+
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/listers/core/v1"
+)
+
+type PodsListerConfig struct {
+	// Namespace can be used to restric the search to a particular namespace.
+	Namespace string
+	// Client is the Kubernetes client.Interface used to build informers.
+	Client kubernetes.Interface
+}
+
+func NewPodsLister(config PodsListerConfig) (v1.PodLister, chan<- struct{}) {
```

Comment:
All informes accept `options ...informers.SharedInformerOption` but none is actually leveraging that, either we include it here as well or we garbage collect them in the other places (I would do so)